### PR TITLE
Add support for cloud-controller-manager in local-up-cluster.sh

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
We need an easy way to test the new external cloud provider. So
let's keep the existing CLOUD_PROVIDER and CLOUD_CONFIG as-is and
add a flag EXTERNAL_CLOUD_PROVIDER to run a separate process.

Since we use hyperkube to run controller-manager, let's add support
for cloud-controller-manager as well to support this use case.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```dev-release-note
Ability to run the cloud-controller-manager from local-up-cluster for easy testing
```

### CODE REVIEW

File: cmd/hyperkube/main.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ func main() {
 	hk.AddServer(NewKubectlServer())
 	hk.AddServer(NewKubeAPIServer())
 	hk.AddServer(NewKubeControllerManager())
+	hk.AddServer(NewCloudControllerManager())
```
Comment (karataliu) [inline cmd/hyperkube/main.go]: seems already added as 'hk.AddAlphaServer(NewCloudControllerManager())' in this file?

Comment (dims) [reply cmd/hyperkube/main.go]: oops! bad merge. thanks!

Comment (dims) [reply cmd/hyperkube/main.go]: i have updated the local-up-cluster.sh too with the "alpha" syntax

**Comment by dims** [comment]
@cblecker ping!

/assign @sttts 
/assign @fejta

@sttts @fejta - fishing for reviews! please take a look when you get a chance.

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -532,9 +533,16 @@ function start_apiserver {
     kube::util::create_client_certkey "${CONTROLPLANE_SUDO}" "${CERT_DIR}" 'client-ca' kube-aggregator system:kube-aggregator system:masters
     kube::util::write_client_kubeconfig "${CONTROLPLANE_SUDO}" "${CERT_DIR}" "${ROOT_CA_FILE}" "${API_HOST}" "${API_SECURE_PORT}" kube-aggregator
 
+    cloud_config=""
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: While it won't directly cause an issue, this creates both an uppercase `CLOUD_CONFIG`, and a lowercase `cloud_config` with very different uses. Makes it less readable/clear. Perhaps "cloud_provider_arg" or something slightly different?

Comment (dims) [reply hack/local-up-cluster.sh]: Thanks fixed!

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -532,9 +533,16 @@ function start_apiserver {
     kube::util::create_client_certkey "${CONTROLPLANE_SUDO}" "${CERT_DIR}" 'client-ca' kube-aggregator system:kube-aggregator system:masters
     kube::util::write_client_kubeconfig "${CONTROLPLANE_SUDO}" "${CERT_DIR}" "${ROOT_CA_FILE}" "${API_HOST}" "${API_SECURE_PORT}" kube-aggregator
 
+    cloud_config_arg=""
```
Comment (sttts) [inline hack/local-up-cluster.sh]: set the "else" case value here. Saves two lines.

File: hack/local-up-cluster.sh [hunk 4]
```diff
@@ -613,13 +626,36 @@ function start_controller_manager {
...
-      --cloud-config="${CLOUD_CONFIG}" \
+      ${cloud_config_arg} \
       --kubeconfig "$CERT_DIR"/controller.kubeconfig \
       --use-service-account-credentials \
       --controllers="${KUBE_CONTROLLERS}" \
       --master="https://${API_HOST}:${API_SECURE_PORT}" >"${CTLRMGR_LOG}" 2>&1 &
     CTLRMGR_PID=$!
+
+    if [[ "${EXTERNAL_CLOUD_PROVIDER:-}" == "true" ]]; then
+        if [ "${CLOUD_CONFIG}" == "" ]; then
```
Comment (sttts) [inline hack/local-up-cluster.sh]: `if [ -z "${CLOUD_CONFIG}" ]; then`

File: hack/local-up-cluster.sh [hunk 5]
```diff
@@ -613,13 +626,36 @@ function start_controller_manager {
...
-      --cloud-provider="${CLOUD_PROVIDER}" \
-      --cloud-config="${CLOUD_CONFIG}" \
+      ${cloud_config_arg} \
       --kubeconfig "$CERT_DIR"/controller.kubeconfig \
       --use-service-account-credentials \
       --controllers="${KUBE_CONTROLLERS}" \
       --master="https://${API_HOST}:${API_SECURE_PORT}" >"${CTLRMGR_LOG}" 2>&1 &
     CTLRMGR_PID=$!
+
+    if [[ "${EXTERNAL_CLOUD_PROVIDER:-}" == "true" ]]; then
```
Comment (sttts) [inline hack/local-up-cluster.sh]: would prefer to see this in a separate start_cloud_controller_manager.

Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh (hunk 4 above)
Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh (hunk 3 above)
Comment (dims) [reply hack/local-up-cluster.sh]: Done!

**Comment by fejta** [comment]
@k8s-mirror-cluster-lifecycle-pr-reviews do we really want to make changes to `local-up-cluster.sh`? If not what is the alternative?

**Comment by sttts** [comment]
> do we really want to make changes to local-up-cluster.sh? If not what is the alternative?

It's the tool many of us (still) use to test code locally. AFAIK there is no alternative, have not even heard of any discussions around that. Especially, kubeadm is no replacement for this developer-only tool.

**Comment by sttts** [comment]
@dims switch the release-note to a dev-release-note. Ptal. I think this is nothing normal users should have to see.

**Comment by dims** [comment]
thanks @sttts

**Pull request merged**